src/output-json-ikev2.c

```diff
@@ -0,0 +1,202 @@
+/* Copyright (C) 2018 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Pierre Chifflier <chifflier@wzdftpd.net>
+ *
+ * Implement JSON/eve logging app-layer IKEv2.
+ */
+
+#include "suricata-common.h"
+#include "debug.h"
+#include "detect.h"
+#include "pkt-var.h"
+#include "conf.h"
+
+#include "threads.h"
+#include "threadvars.h"
+#include "tm-threads.h"
+
+#include "util-unittest.h"
+#include "util-buffer.h"
+#include "util-debug.h"
+#include "util-byte.h"
+
+#include "output.h"
+#include "output-json.h"
+
+#include "app-layer.h"
+#include "app-layer-parser.h"
+
+#include "app-layer-ikev2.h"
+#include "output-json-ikev2.h"
+
+#ifdef HAVE_RUST
+#ifdef HAVE_LIBJANSSON
+
+#include "rust.h"
+#include "rust-ikev2-log-gen.h"
+
+typedef struct LogIKEv2FileCtx_ {
+    LogFileCtx *file_ctx;
+    uint32_t    flags;
+    bool        include_metadata;
+} LogIKEv2FileCtx;
+
+typedef struct LogIKEv2LogThread_ {
+    LogIKEv2FileCtx *ikev2log_ctx;
+    uint32_t            count;
+    MemBuffer          *buffer;
+} LogIKEv2LogThread;
+
+static int JsonIKEv2Logger(ThreadVars *tv, void *thread_data,
+    const Packet *p, Flow *f, void *state, void *tx, uint64_t tx_id)
+{
+    IKEV2Transaction *ikev2tx = tx;
+    LogIKEv2LogThread *thread = thread_data;
+    json_t *js, *ikev2js;
+
+    js = CreateJSONHeader((Packet *)p, 0, "ikev2");
```

Comment:
this has been updated in master. See https://github.com/OISF/suricata/commit/710c7b821fcb919dfaa816f45a8b342acee622f4

Reply:
OK, fixed